### PR TITLE
Feature :: Button group widget

### LINKED ISSUE #2876: Feature :: support for button group widget
### Is your proposal related to a problem?
Currently tooljet doesn't have a button group widget.

### Describe the solution you'd like
Add button group widget.

### If the feature is approved, would you be willing to submit a PR?
Yes.

### PR DESCRIPTION
**Pull request by stepinfwd**
closes #2876

### CODE REVIEW

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Widget works fine!, few things:

Since you're using an array as exposed variable for both single selection and multiple selection, please ensure that the initially exposed value is `[1]` and not `1`.

Also please add documentation, sync with @withshubh if that helps.

**Comment by stepinfwd** [comment]
updated with required changes @sherfin94

**Comment by sherfin94** [comment]
Can we make it so, that we can supply default selected an actual array instead of a string that will be parsed as an array?
image

**Comment by sherfin94** [comment]
When the `Enable multiple selection` property is toggled on or off, the widget's state (the buttons that are displayed as selected, and the exposed value) should reset to the `default selected` value. Presently, the widget stays as if no buttons are selected and the exposed value remains the same.

File: frontend/src/Editor/Components/components.js [hunk 1]
```diff
@@ -2122,6 +2182,71 @@ ReactDOM.render(<ConnectedComponent />, document.body);`,
...
+    exposedVariables: {
+      selected: [1],
+    },
+    definition: {
+      others: {
+        showOnDesktop: { value: '{{true}}' },
+        showOnMobile: { value: '{{false}}' },
+      },
+      properties: {
+        label: { value: `Button` },
```
Comment (sherfin94) [inline frontend/src/Editor/Components/components.js:2232]:
```suggestion
        label: { value: `Button group` },
```

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Suggested few changes, otherwise looks fine.

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Final change, there seems to be unnecessary padding on the top when button group label is removed:

image

If header is removed, this padding is not required.

**Review by sherfin94 — APPROVED** [review]

**Pull request merged**